domain/src/main/java/org/sagebionetworks/research/domain/step/interfaces/Step.java

```diff
@@ -67,4 +68,12 @@
      */
     @NonNull
     String getType();
+
+    /**
+     * Returns a Result for this step.
+     *
+     * @return a Result for this step.
+     */
+    @NonNull
+    Result instantiateStepResult();
```

Comment:
This is mirroring iOS where steps produce their own result. I also agree however that it would be better if steps didn't instantiate their own results.